core/src/main/scala/kafka/server/KafkaConfig.scala

```diff
@@ -636,6 +605,15 @@ class KafkaConfig private(doLog: Boolean, val props: util.Map[_, _])
     if (nodeId != brokerId) {
       throw new ConfigException(s"You must set `${KRaftConfigs.NODE_ID_CONFIG}` to the same value as `${ServerConfigs.BROKER_ID_CONFIG}`.")
     }
+    if (processRoles.isEmpty) {
+      throw new ConfigException(s"You must set `${KRaftConfigs.PROCESS_ROLES_CONFIG}`. ZooKeeper mode is not supported.")
+    } else {
+      // KRaft-based metadata quorum
+      if (nodeId < 0) {
+        throw new ConfigException(s"Missing configuration `${KRaftConfigs.NODE_ID_CONFIG}` which is required " +
```

Comment:
We also made this mandatory. So, the message should probably reflect that.

Reply:
It looks like it's defined with `atLeast(0)` so we can remove this.